### ISSUE TITLE
Incorrect git diff file generated by plugin

### ISSUE THREAD

**Issue by grsky360** [form-com/diff-coverage-gradle#34]
When I'm using default git.compare, I found the diff file is not completely correct.
For example, we have master and dev branchs.
1. master have some commit that dev don't have.
2. dev have some commit that master don't have.
3. We're on the dev branch.
4. Diff coverage comparing to master branch.

In this case, the diff file is including the commits on master and treating as delete operation.

Actually, the master should be always latest comparing to our develop branch, and have some commits we don't have if we don't merge master.
I'm not sure it's a issue or you do want this, but I think we should ignore the commits dev don't have, at least provide a config value.

For now, I use this command to generate the diffFile, please check and hope it helps.
```
1. git diff origin/master... > reports/git.diff
2. git diff --ignore-space-change origin/master... > reports/git.diff
```

**Comment by grsky360** [comment]
I just found a discussion about this, hope it could help
https://www.eclipse.org/forums/index.php/t/1065862/

**Comment by SurpSG** [comment]
Hi @grsky360 
Thank you for reporting the issue. Working on it

**Comment by grsky360** [comment]
May I allowed to ask the rogress of this fix, and is there any snapshot version usable?

**Comment by SurpSG** [comment]
Unfortunatelly, I didn't have enough time to fix the issue. Also, fixing of the issue is not trivial. I need much more deep diving into JGit. The [solution](https://github.com/form-com/diff-coverage-gradle/issues/34#issuecomment-968205849) you provided works fine, but it doesn't include uncommited changes that is very useful for local runs.

If the issue is critical for you then you can apply some workarounds:
* provide diff from your CI if possible 
```groovy
diffCoverageReport {
      diffSource.url = 'http://diff-location'
      // or
      diffSource.file = 'path/to/diff'
}
```
* generate diff with git
```groovy
    createDiffUrl = { ->
        // obtain diff base from property if any or default HEAD
        // usage: ./gradlew diffCoverage -PdiffBase=master
        def diffBase = project.hasProperty('diffBase') ? project.diffBase : 'HEAD'
       
        // create a temp file
        def file = Files.createTempFile(URLEncoder.encode(project.name, 'UTF-8'), '.diff').toFile()
        // generate diff by git and write the content to the temp file
        file.withOutputStream { out ->
            exec {
                commandLine 'git', 'diff', diffBase
                standardOutput = out
            }
        }

        // build url to the file. Or return path to the file if u like
        return file.toURI().toURL()
    }

    diffCoverageReport {
       diffSource.url = createDiffUrl()
    }
```

I'll try to fix the issue ASAP

**Comment by grsky360** [comment]
Ok, I'll try to fix it manually by file based diff file. Thank you!

**Comment by SurpSG** [comment]
fixed by #34

Planned release 0.9.1
